api/wasm.go

```diff
@@ -366,6 +366,20 @@ type Function interface {
 	// the end-to-end demonstrations of how these terminations can be performed.
 	Call(ctx context.Context, params ...uint64) ([]uint64, error)
 
+	// CallWithStack is an optimized variation of Call that saves memory
+	// allocations when the stack slice is reused across calls.
+	//
+	// Stack length must be at least the max of parameter or result length.
```

Comment:
I'd love to see an example prior to merge.

```go
// # Example
// ...
```

Reply:
Not forgotten. Just thinking about a good example that passes the message with only "local" _context_ (to reduce cognitive load).

Thinking about what @anuraaga said, maybe the simplest way to transmit the idea without resorting to global variables, is this: a host calling a guest function 100 times in a loop, and reusing the stack for (just) those 100 calls. Like matching text against 100 regexes.

Wdyt?

Reply:
sounds good, sounds something like a index correct version of this.

```
stack = ...
for i, search := range searchParams {

// copy the next params to the stack
copy(stack, search)
if err := searchFn.CallWithStack(ctx, stack) {
return err
} else if stack[0] == 1 { // found
return i // searchParams[i] matched!
}
}
```